src/org/zaproxy/zap/extension/backslashpoweredscanner/AttributeAnalysis.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2017 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.backslashpoweredscanner;
+
+import java.util.HashMap;
+
+import org.parosproxy.paros.network.HttpHeaderField;
+import org.parosproxy.paros.network.HttpResponseHeader;
+
+/*
+ * AttributeAnalysis analyzes a bundle of HTTP response attribute (currently only headers, status cod, and content type)
+ *
+ */
+public class AttributeAnalysis {
+    private HashMap<String, Object> invariantAttributes;
+    private HashMap<String, String> invariantHeaders;
+
+    public AttributeAnalysis() {
+        this.invariantAttributes = new HashMap<>();
+        this.invariantHeaders = new HashMap<>();
+    }
+
+    public AttributeAnalysis(HttpResponseHeader header) {
+        this.invariantHeaders = new HashMap<>();
+        for (HttpHeaderField h : header.getHeaders()) {
+            invariantHeaders.put(h.getName(), h.getValue());
+        }
+        this.invariantAttributes.put("headers", invariantHeaders);

```

Comment:
This leads to an NPE as `invariantAttributes` is `null` when the `put` is attempted:
```
81596 [ZAP-ActiveScanner-0] ERROR org.zaproxy.zap.extension.backslashpoweredscanner.BackendParameterEnumeration  - Error occurred while scanning a message:
java.lang.NullPointerException
	at org.zaproxy.zap.extension.backslashpoweredscanner.AttributeAnalysis.<init>(AttributeAnalysis.java:45)
	at org.zaproxy.zap.extension.backslashpoweredscanner.Attack.<init>(Attack.java:64)
	at org.zaproxy.zap.extension.backslashpoweredscanner.BackSlashPoweredAbstractAppParamPlugin$PayloadInjector.buildAttack(BackSlashPoweredAbstractAppParamPlugin.java:113)
	at org.zaproxy.zap.extension.backslashpoweredscanner.BackendParameterEnumeration.scan(BackendParameterEnumeration.java:90)
	at org.parosproxy.paros.core.scanner.AbstractAppParamPlugin.scan(AbstractAppParamPlugin.java:263)
	at org.parosproxy.paros.core.scanner.AbstractAppParamPlugin.scanVariant(AbstractAppParamPlugin.java:191)
	at org.parosproxy.paros.core.scanner.AbstractAppParamPlugin.scan(AbstractAppParamPlugin.java:162)
	at org.parosproxy.paros.core.scanner.AbstractPlugin.run(AbstractPlugin.java:338)
	at java.lang.Thread.run(Thread.java:748)
```

If that NPE is handled via initialization (null check then initialize if needed) then some scanning can occur but the following NPE pops:
```
1073671 [ZAP-ActiveScanner-0] ERROR org.zaproxy.zap.extension.backslashpoweredscanner.BackendParameterEnumeration  - Error occurred while scanning a message:
java.lang.NullPointerException
	at org.zaproxy.zap.extension.backslashpoweredscanner.Attack.updateWith(Attack.java:86)
	at org.zaproxy.zap.extension.backslashpoweredscanner.BackSlashPoweredAbstractAppParamPlugin$PayloadInjector.verify(BackSlashPoweredAbstractAppParamPlugin.java:131)
	at org.zaproxy.zap.extension.backslashpoweredscanner.BackSlashPoweredAbstractAppParamPlugin$PayloadInjector.fuzz(BackSlashPoweredAbstractAppParamPlugin.java:60)
	at org.zaproxy.zap.extension.backslashpoweredscanner.BackendParameterEnumeration.scan(BackendParameterEnumeration.java:105)
	at org.parosproxy.paros.core.scanner.AbstractAppParamPlugin.scan(AbstractAppParamPlugin.java:263)
	at org.parosproxy.paros.core.scanner.AbstractAppParamPlugin.scanVariant(AbstractAppParamPlugin.java:191)
	at org.parosproxy.paros.core.scanner.AbstractAppParamPlugin.scan(AbstractAppParamPlugin.java:162)
	at org.parosproxy.paros.core.scanner.AbstractPlugin.run(AbstractPlugin.java:338)
	at java.lang.Thread.run(Thread.java:748)
```